persist/fs/commitlog/commit_log_test.go

```diff
@@ -330,6 +330,43 @@ func TestCommitLogWrite(t *testing.T) {
 	assertCommitLogWritesByIterating(t, commitLog, writes)
 }
 
+func TestCommitLogReaderIsNotReusable(t *testing.T) {
+	opts, scope := newTestOptions(t, overrides{
+		strategy: StrategyWriteWait,
+	})
+	defer cleanup(t, opts)
+
+	commitLog := newTestCommitLog(t, opts)
+
+	writes := []testWrite{
+		{testSeries(0, "foo.bar", 127), time.Now(), 123.456, xtime.Second, []byte{1, 2, 3}, nil},
+		{testSeries(1, "foo.baz", 150), time.Now(), 456.789, xtime.Second, nil, nil},
+	}
+
+	// Call write sync
+	writeCommitLogs(t, scope, commitLog, writes).Wait()
+
+	// Close the commit log and consequently flush
+	assert.NoError(t, commitLog.Close())
+
+	// Assert writes occurred by reading the commit log
+	assertCommitLogWritesByIterating(t, commitLog, writes)
+
+	// Assert commitlog file exists and retrieve path
+	fsopts := opts.FilesystemOptions()
+	files, err := fs.CommitLogFiles(fs.CommitLogsDirPath(fsopts.FilePathPrefix()))
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(files))
+
+	// Assert commitlog cannot be opened more than once
```

Comment:
maybe break the following lines into a separate unit test for better isolation

Reply:
We already have a unit test for all the previous lines, but I needed to setup the files for bootstrapping and bootstrap them once before I could test trying to bootstrap them again

Reply:
You're creating a new reader below as is, why does it matter if there's data on disk (i.e. all the preceeding lines) before you check reuse of the commit log reader?

Reply:
Yeah I guess you're right it doesn't matter, I'll remove some of that stuff.

Reply:
Sounds good

Reply:
oh actually part of it was just so I would have a file to read